README.md

```diff
@@ -1,120 +1,87 @@
-
-### Building with Nix
-
-```bash
-# install Nix package manager:
-$ curl https://nixos.org/nix/install | sh
-
-# run from root of the project folder (`Astar/` folder)
-$ nix-shell -I nixpkgs=channel:nixos-21.05 third-party/nix/shell.nix --run "cargo build --release"
-```
-
-## Running a Collator Node
-
-To set up a collator node, you must have a fully synced node with the proper arguments, which can be done with the following command.
-
-```bash
-# start the Shiden collator node with
-$ ./target/release/astar-collator \
-  --base-path <path to save blocks> \
-  --name <node display name> \
-  --port 30333 \
-  --ws-port 9944 \
-  --rpc-port 9933 \
-  --telemetry-url 'wss://telemetry.polkadot.io/submit/ 0' \
-  --rpc-cors all \
-  --collator
+pub type PSP34Transactor = PSP34MutateAdapter<
+    // Use this non-fungibles implementation:
+    Contracts,
+    // Use this currency when it is a non-fungible asset matching the given location or name:
+    ConvertedConcreteAssetId<ClassId, InstanceId, AstarPSP34ClassIdConverter, AstarPSP34InstanceIdConverter>,
+    // Convert an XCM MultiLocation into a local account id:
+    LocationToAccountId,
+    // Our chain's account ID type (we can't get away without mentioning it explicitly):
+    AccountId,
+    // We don't track any teleports of `Assets`.
+    Nothing,
+    // We don't track any teleports of `Assets`.
+    CheckingAccount,
+>;
 ```
+Trait definition for PSP34MutateAdapter can be found here-> https://github.com/gitofdeepanshu/polkadot/pull/1
 
-Now, you can obtain the node's session key by sending the following RPC payload.
+## Trait Bounds for Pallet Contracts
+Structure of our new Adapter
 
-```bash
-# send `rotate_keys` request
-$ curl -H 'Content-Type: application/json' --data '{ "jsonrpc":"2.0", "method":"author_rotateKeys", "id":1 }' localhost:9933
-
-# should return a long string of hex, which is your session key
-{"jsonrpc":"2.0","result":"<session key in hex>","id":1}
+```
+impl<
+		Assets: nonfungibles::MutatePSP34<AccountId>,
+		Matcher: MatchesNonFungibles<Assets::CollectionId, Assets::ItemId>,
+		AccountIdConverter: Convert<MultiLocation, AccountId>,
+		AccountId: Clone + Eq + From<[u8;32]>, // can't get away without it since Currency is generic over it.
+		CheckAsset: AssetChecking<Assets::CollectionId>,
+		CheckingAccount: Get<Option<AccountId>>,
+	>
+	PSP34MutateAdapter<
+		Assets,
+		Matcher,
+		AccountIdConverter,
+		AccountId,
+		CheckAsset,
+		CheckingAccount,
+	>
 ```
 
-After this step, you should have a validator node online with a session key for your node.
-For key management and validator rewards, consult our [validator guide online](https://docs.astar.network/build/validator-guide/configure-node).
+In order to make pallet contracts compatible with our new adapter, we have to implement trait `nonfungibles::MutatePSP34<AccountId>` for pallet_contracts.
+Here is the PR for new trait definitions for pallet_contracts: https://github.com/gitofdeepanshu/substrate/pull/1
```

Comment:
Left a comment about it below. But in general, you are correct, some form of support is required.